cmd/wallet-sdk-gomobile/wrapper/error.go

```diff
@@ -42,6 +42,7 @@ func ToMobileErrorWithTrace(err error, trace *otel.Trace) error {
 			Category: walletError.Scenario,
 			Details:  walletError.ParentError,
 			TraceID:  traceID,
+			Raw:      err.Error(),
```

Comment:
'Raw' name is a little bit unclear. And also it looks like invalid usage of our errors model,  walleterror.Error shouldn't be wrapped just with a simple error. Original walleterror.Error should be handled, and new walleterror.Error produced.

Reply:
In case when a top-level error is not walleterror.Error, maybe it will be good to treat this as an error, or even panic. Because this means that we just done an error in the wallet-sdk code.

Reply:
I see some challenges here...

Many of the error codes that come from Wallet-SDK happen at a low-level (i.e. in nested functions). In Go, it's quite beneficial to wrap errors from lower-level functions, which helps provide more information about the source of an error for debugging purposes. So, if I were to only pass through the low-level error without wrapping, we're losing potentially quite a lot of contextual information. Also, keep in mind that with the new use-cases and flows we're supporting (e.g. Wallet-initiated flow, pre-auth, auth, different authorization schemes), theres more and more places where the same low-level functions are called from higher-level functions. So, there will be a lot of places where we have to ensure we never wrap errors (and lots of good context we're going to lose).

With that in mind, this will be challenging to maintain, as this becomes an "implicit" restriction of Wallet-SDK development, which will be difficult to maintain. Note that with our current error code structure, we've already introduced one implicit restriction - where you need to ensure that you never wrap a walleterror.Error with another walleterror.Error, since that causes the lower-level walleterror.Error to be "unseen" by the gomobile parse function... we should keep these implicit restrictions to a minimum.

So, my idea with this "Raw" field was to try to get the best of both worlds - allow for error code checks by the calling code, while also providing a way to ensure that extra higher-level contextual error messages can be seen as well, if desired.

Do you think there's another way we can achieve this "best of both worlds" solution I'm looking for, without minimizing implicit restrictions on how errors are handled in Go code?

P.S. Perhaps "Full" would be a better name than "Raw"?

Reply:
Yes, "Full" would be better than Raw

Reply:
In this case, only one implicit restriction is left, which is always return walleterror.Error from gomobile functions that are accessible from mobile code. We can also make this restriction a little bit stronger. By creating helper functions for tests, that will not only check that error has the correct message but also that it is walleterror.Error, with correct Code, and Category

Reply:
Just realized here one other case. When we don't want to change the nature of the error, but just add additional trace information, so we know that it happened when call from a particular function. As I understand you mean this case. 
in this case, we can replace fmt.Errorf with a more clever function. That will keep the original walleterror.Error a top, but just add additional info at the beginning of the details filed of that original error.

Reply:
> Just realized here one other case. When we don't want to change the nature of the error, but just add additional trace information, so we know that it happened when call from a particular function. As I understand you mean this case. in this case, we can replace fmt.Errorf with a more clever function. That will keep the original walleterror.Error a top, but just add additional info at the beginning of the details filed of that original error.

Ah, I think I understand what you're saying. Let me look into this and see...

> I this it is good that gomobile parse function will not see lower-level walleterror.Error, but just see high-level walleterror.Error. In this case, "details" field will contain low-level error descriptions. So it can be used to debug error couses. WalletSDK users should react to higher-level errors. Low-level error in this case is something that depends on implementation and can be changed with implementation change.

I think based on your most recent comment, you might have realized this already, but just in case: it's actually the higher-level error details that were getting lost (rather than lower-level ones). Actually, all the lower-level details do get sent - if we want to hide them, that would be a different thing....

Reply:
> Do you think there's another way we can achieve this "best of both worlds" solution I'm looking for, without minimizing implicit restrictions on how errors are handled in Go code?

Oops... meant to say _while_ minimizing implicit restrictions, not _without_.

Reply:
I'm thinking I can do this in a followup PR, since this one is already pretty big. Does that sound ok?